File: OpenEdXMobile/src/main/java/org/edx/mobile/view/OfflineSupportUtils.java
package org.edx.mobile.view;


import android.app.Activity;
import androidx.annotation.Nullable;

import org.edx.mobile.base.RoboAppCompatActivity;
import org.edx.mobile.interfaces.SnackbarStatusListener;
import org.edx.mobile.util.NetworkUtil;

public class OfflineSupportUtils {
    public static void setUserVisibleHint(@Nullable Activity activity, boolean isVisibleToUser,
                                          boolean isShowingFullScreenError) {
        if (isVisibleToUser &&
                activity != null &&
                activity instanceof SnackbarStatusListener) {
            ((SnackbarStatusListener) activity).resetSnackbarVisibility(isShowingFullScreenError);
        }
    }

    public static void onRevisit(@Nullable Activity activity) {
        if (activity != null &&
                activity instanceof SnackbarStatusListener &&
                NetworkUtil.isConnected(activity)) {
            ((SnackbarStatusListener) activity).hideSnackBar();
        }
    }

    public static void onNetworkConnectivityChangeEvent(@Nullable Activity activity,
                                                        boolean isVisibleToUser,
                                                        boolean isShowingFullScreenError) {

        /*
         This event is consumed even when the activity isn't in foreground.
         We need to ensure that the following logic only executes when the activity and fragment
         are in the foreground.
         */
        final boolean isActivityVisible;
        {
            if (activity != null && activity instanceof RoboAppCompatActivity) {
                isActivityVisible = ((RoboAppCompatActivity) activity).isInForeground();
            } else {
                isActivityVisible = false;
            }
        }
        if (isActivityVisible &&
                isVisibleToUser &&
                activity instanceof SnackbarStatusListener &&
                !NetworkUtil.isConnected(activity)) {
            ((SnackbarStatusListener) activity).resetSnackbarVisibility(isShowingFullScreenError);
        }
    }
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/view/OfflineSupportUtils.kt
package org.edx.mobile.view

import android.app.Activity
import org.edx.mobile.base.BaseAppActivity
import org.edx.mobile.interfaces.SnackbarStatusListener
import org.edx.mobile.util.NetworkUtil

object OfflineSupportUtils {
    @JvmStatic
    fun setUserVisibleHint(
        activity: Activity?, isVisibleToUser: Boolean,
        isShowingFullScreenError: Boolean
    ) {
        if (isVisibleToUser &&
            activity is SnackbarStatusListener
        ) {
            (activity as SnackbarStatusListener).resetSnackbarVisibility(isShowingFullScreenError)
        }
    }

    @JvmStatic
    fun onRevisit(activity: Activity?) {
        if (activity is SnackbarStatusListener &&
            NetworkUtil.isConnected(activity)
        ) {
            (activity as SnackbarStatusListener).hideSnackBar()
        }
    }

    @JvmStatic
    fun onNetworkConnectivityChangeEvent(
        activity: Activity?,
        isVisibleToUser: Boolean,
        isShowingFullScreenError: Boolean
    ) {

        /*
         This event is consumed even when the activity isn't in foreground.
         We need to ensure that the following logic only executes when the activity and fragment
         are in the foreground.
         */
        val isActivityVisible: Boolean
        run {
            isActivityVisible = if (activity is BaseAppActivity) {
                activity.isInForeground
            } else {
                false
            }
        }
        if (isActivityVisible &&
            isVisibleToUser &&
            activity is SnackbarStatusListener &&
            !NetworkUtil.isConnected(activity)
        ) {
            (activity as SnackbarStatusListener).resetSnackbarVisibility(isShowingFullScreenError)
        }
    }
}
